src/server/components/notebook-actions-menu.jsx

```diff
@@ -9,7 +9,7 @@ import {
   selectFileAndFormatMetadata,
```

Comment:
this component needs help :-(

i'd originally wanted to suggest that we fix it right now, but i don't want to slow down the progress on the file upload API.

so rather than stopping to fix this now, please add this FIXME in this file:

FIXME this component needs to be refactored to be much dumber than it currently is. (1) it does not need to have internal state (2) it should just dispatch thunks that call `selectFileAndFormatMetadata`, `uploadFile`, and `updateFile` and handle the promises/`.then`s from those those API calls (3) it should not import those functions directly, it should import the action creators that wrap them, and those actions should be passed in via mapDispatchToProps

(or, i guess it's up to you... if you want to tackle this now i don't think it would actually take that long and it would be a fine use of time -- as our codebase grows in size and complexity, we'll need to aim for more architectural scalability, which will mean cleaning up things like this being stricter about our interfaces, so this would be a reasonable thing to spend an hour on towards that end)

Reply:
These are server components, not client ones, so this isn't relevant. Since our server components are an afterthought to the rest of the app, I'd vote for not refactoring anything in this PR unless we prioritize actually designing the server components.